### PR TITLE
SOLR-16616: JWTAuthPlugin: Read trusted X509 certificates from multi files

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-16616

A concrete use case for this is in a Kubernetes environment where a configMap exists with two `.cer` files, one for the SSL certificate of KeyCloak, and another file with SSL cert for the private CA. Currently we have been copying and concatenating the string content of those files into the `trustedCerts` configuration key. But that is cumbersome when you have several k8s clusters with different certificates for each.

With this feature, we'll supply a both files as this
```json
"trustedCertsFile": ["/path/to/hostCertFile.pem", "/path/to/caCertFile.pem"]
```

I considered adding a new parameter `trustedCertsFiles`, but went for accepting list on the existing one.

### CODE REVIEW

File: solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java [hunk 1]
```diff
@@ -297,6 +283,54 @@ public void init(Map<String, Object> pluginConfig) {
...
+    Collection<X509Certificate> certs = new HashSet<>();
+    if (trustedCertsFileObj instanceof String) {
+      String trustedCertsFile = (String) trustedCertsFileObj;
+      log.info("Reading trustedCerts from file {}", trustedCertsFile);
+      try {
+        certs.addAll(parseCertsFromFile(trustedCertsFile));
+      } catch (IOException e) {
+        throw new SolrException(
+            SolrException.ErrorCode.SERVER_ERROR, "Failed to read file " + trustedCertsFile, e);
+      }
```
Comment (risdenk) [inline solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java:301]: Couldn't you just make this string into a List of one and use the same logic as the list? That would simplify and remove the duplications.

File: solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java [hunk 2]
```diff
@@ -297,6 +283,54 @@ public void init(Map<String, Object> pluginConfig) {
     lastInitTime = Instant.now();
   }
 
+  /**
+   * Given a configuration object of a file name or list of file names, read X509 certificates from
+   * each file
+   */
+  @SuppressWarnings("unchecked")
```
Comment (risdenk) [inline solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java:290]: What is unchecked here?

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java (hunk 2 above)
Comment (janhoy) [reply solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java:290]: Unchecked cast from `Object` to `List<String>`

File: solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java (hunk 1 above)
Comment (janhoy) [reply solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java:301]: I had the same thought on the train home from work today :) Will investigate.

File: solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java [hunk 3]
```diff
@@ -297,6 +283,54 @@ public void init(Map<String, Object> pluginConfig) {
...
+    return certs;
+  }
+
+  /**
+   * Given a filename string, validate the file and then read any X509 certificates from it
+   *
+   * @return list of certificates found in file
+   */
+  Collection<? extends X509Certificate> parseCertsFromFile(String certFileName) throws IOException {
+    Path certFilePath = Paths.get(certFileName);
```
[the remark this answers is not in the document]
Comment (janhoy) [reply solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java:327]: @sonatype-lift ignore
We check in the line below that this path is safe.

**Review by risdenk — APPROVED** [review]

**Comment by janhoy** [comment]
Thanks for the review. I'll merge this on Friday if no other comments.

**Pull request merged**